sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -151,7 +151,8 @@ def create_job_resources(options,  # type: PipelineOptions
     setup_options = options.view_as(SetupOptions)
 
     # Stage a requirements file if present.
-    if setup_options.requirements_file is not None:
+    if (setup_options.requirements_file is not None and
```

Comment:
the artifact infos are generated from the result of stager create_resources, so it is easiest to reduce it in the stager, also this means we don't re-create or re-download unnecessary resources.

Reply:
It may require some refactoring, but I think the over-all result would be cleaner. (Maybe this is also messy due to a partial transition in how Dataflow handles artifacts compared to every other runner...) 

However, if you prefer to keep it like this that's OK.